docs/features/nullable-reference-types.md

```diff
@@ -93,6 +93,9 @@ Invocation of methods annotated with the following attributes will also affect f
 - `[EnsuresNotNull]` (e.g. `ThrowIfNull`)
 - `[AssertsTrue]` (e.g. `Debug.Assert`) and `[AssertsFalse]`
 
+## Unreachable code
+Expressions that produce an unreachable state produce a not-null r-value and an oblivious l-value.
+
```

Comment:
I don't think this is correct.  Reachability only affects the flow state.  It doesn't affect the nullable annotation except to the extent that it is computed from the flow state by type inference.

Reply:
You're stating a conclusion, but not a rationale. Do you see a conceptual or practical downside to this design? Or are you just saying this statement is not currently included in the spec?

Reply:
We can discuss more when I return, but I think this leads to an inconsistent design.  Generally, type inference flows up from subexpressions to enclosing expressions independent of reachability.  Nullable flow state, on the other hand, flows based on reachability.  Those intersect only at type inference, where an rvalue is input to type inference and results in a computed nullable annotation.  The spec doesn't say what you suggest, and I think the spec *should not* say it.  It is a change in direction that I think we would have to discuss in detail.  If we were to take this position, then a ?: expression, when only one operand were reachable, would take the entire type from that operand and not require convertability from the type of the other operand to the result type.  But that is not how the types work.  Generally, what you suggest is inconsistent with the way the types flow in initial binding.

---
In reply to: [268482843](https://github.com/dotnet/roslyn/pull/34323#discussion_r268482843) [](ancestors = 268482843)